data-prepper-logstash-configuration/src/test/java/org/opensearch/dataprepper/logstash/parser/LogstashVisitorTest.java

```diff
@@ -0,0 +1,559 @@
+package org.opensearch.dataprepper.logstash.parser;
+
+import org.antlr.v4.runtime.tree.TerminalNodeImpl;
+import org.junit.jupiter.api.Test;
+import static org.hamcrest.CoreMatchers.equalTo;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.junit.Assert.assertThrows;
+import static org.mockito.BDDMockito.given;
+import static org.mockito.Mockito.spy;
+import static org.mockito.Mockito.mock;
+
+import org.mockito.Mockito;
+import org.opensearch.dataprepper.logstash.LogstashParser;
+import org.opensearch.dataprepper.logstash.exception.LogstashParsingException;
+import org.opensearch.dataprepper.logstash.model.LogstashConfiguration;
+import org.opensearch.dataprepper.logstash.model.LogstashPlugin;
+import org.opensearch.dataprepper.logstash.model.LogstashPluginType;
+import org.opensearch.dataprepper.logstash.model.LogstashAttribute;
+
+import java.util.Collections;
+import java.util.Arrays;
+import java.util.List;
+import java.util.LinkedList;
+import java.util.Map;
+
+class LogstashVisitorTest {
+
+    private LogstashVisitor createObjectUnderTest() {
+        return spy(new LogstashVisitor());
+    }
+
+    @Test
+    void visit_config_test() {
+        final LogstashParser.ConfigContext configContextMock = mock(LogstashParser.ConfigContext.class);
+        final LogstashParser.Plugin_sectionContext pluginSectionMock = mock(LogstashParser.Plugin_sectionContext.class);
+        final LogstashParser.Plugin_typeContext pluginTypeContextMock = mock(LogstashParser.Plugin_typeContext.class);
+        final LogstashParser.Branch_or_pluginContext branchOrPluginContextMock = mock(LogstashParser.Branch_or_pluginContext.class);
+
+        given(configContextMock.plugin_section()).willReturn(Collections.singletonList(pluginSectionMock));
+        given(pluginSectionMock.plugin_type()).willReturn(pluginTypeContextMock);
+        given(pluginTypeContextMock.getText()).willReturn("input");
+        given(pluginSectionMock.branch_or_plugin()).willReturn(Collections.singletonList(branchOrPluginContextMock));
+
+        LogstashVisitor logstashVisitor = createObjectUnderTest();
+        Mockito.doReturn(TestDataProvider.pluginWithOneArrayContextAttributeData()).when(logstashVisitor).visitBranch_or_plugin(branchOrPluginContextMock);
+
+        LogstashConfiguration actualLogstashConfiguration = (LogstashConfiguration) logstashVisitor.visitConfig(configContextMock);
+        LogstashConfiguration expectedLogstashConfiguration = TestDataProvider.configData();
+
+        assertThat(actualLogstashConfiguration.getPluginSection(LogstashPluginType.INPUT).size(),
+                equalTo(expectedLogstashConfiguration.getPluginSection(LogstashPluginType.INPUT).size()));
+        Mockito.verify(logstashVisitor, Mockito.times(1)).visitPlugin_section(pluginSectionMock);
+    }
```

Comment:
It appears you are mocking the same objects over and over again in each test. I would encourage you to use `@Mock` annotation and create a class variable to reuse. I would also encourage you to create a LogstashVisitor class variable to assign in a `@BeforeEach` function as well. This will help reduce the duplicate code you have.